app/models/comment.rb

```diff
@@ -647,83 +627,30 @@ def self.recent_threads(user)
       .pluck(:thread_id)
     return Comment.none if thread_ids.empty?
 
-    Comment
-      .joins(<<~SQL
-        inner join (
-          with recursive discussion as (
-          select
-            c.id,
-            0 as depth,
-            (select count(*) from comments where parent_comment_id = c.id) as reply_count,
-            cast(confidence_order as char(#{Comment::COP_LENGTH}) character set binary) as confidence_order_path
-            from comments c
-            where
-              thread_id in (#{thread_ids.join(", ")}) and
-              parent_comment_id is null
-          union all
-          select
-            c.id,
-            discussion.depth + 1,
-            (select count(*) from comments where parent_comment_id = c.id),
-            cast(concat(
-              left(discussion.confidence_order_path, 3 * (depth + 1)),
-              c.confidence_order
-            ) as char(#{Comment::COP_LENGTH}) character set binary)
```

Comment:
It will be nice to simplify the definition of confidence_order_path from confidence_order.

Reply:
Yes! This is a [big evil performance hack](https://push.cx/stream/2024-09-05-you-re-allowed-to-be-evil-in-performance-code) for our biggest hot path feature because MariaDB didn't allow me to use arrays in an RCTE.